### PR TITLE
[civ2] migrate core, serve, serverless tests to civ2

### PR DESCRIPTION
**Pull request by can-anyscale**
Migrate core, serve and serverless tests to civ2
- Move test runtime builder to wanda to take advantage of cache, one for each team (corebuild, servebuild, serverlessbuild). This is better than do it after ray installation, because otherwise that layer is not cachable.
- Build ray, and only ray, right before running the tests. ci/ray_ci/tests.env.Dockerfile: basically the minimal of build.Docker part.
- There are a lot of files, they are all just configurations (or moving things around)

Test:
- CI
- tests finished in 40min (previously 2 hours), all green

### CODE REVIEW

**Comment by can-anyscale** [comment]
Screenshot 2023-08-18 at 4 58 31 PM

File: .buildkite/forge.rayci.yml [hunk 1]
```diff
@@ -13,3 +13,15 @@ steps:
...
+    wanda: ci/docker/core.build.wanda.yaml
+    depends_on: oss-ci-base_build
+
+  - name: servebuild
+    wanda: ci/docker/serve.build.wanda.yaml
+    depends_on: oss-ci-base_build
+
+  - name: serverlessbuild
+    wanda: ci/docker/serverless.build.wanda.yaml
+    depends_on: oss-ci-base_build
```
Comment (aslonnie) [inline .buildkite/forge.rayci.yml:27]: you actually diverged and did not use the `copy_everything` (though it is available already). this is awesome! great job!

so it seems that the `ci_build` image is actually not required for these tests to run?

could you split core and serve into two different PR's and also ask one or more eng from core / serve team to review?

**Review by aslonnie — COMMENTED** [review]
if you can split the PR into two it would be nice, but seems that it would need a transitional period for the config files of `ray_ci` scripts.

so I will give you the approval here. but maybe make sure some core/serve engs are in the loop and knows that their tests have been moved and becomes blocking.

**Review by aslonnie — APPROVED** [review]

**Comment by aslonnie** [comment]
@scottjlee , your sharding on civ2. thanks to @can-anyscale 's great work.

**Comment by can-anyscale** [comment]
@aslonnie w00h00, this PR makes the test runner no longer compatible with civ01 so I need to migrate all tests over in one PR

@xieus, @rickyyx , @edoakes : this PR moves some of the core + serve tests to civ02, make it blocking, as well as runs 2-3x faster

**Comment by aslonnie** [comment]
> runs 2-3x faster

will they run even faster with more sharding?

**Comment by can-anyscale** [comment]
> will they run even faster with more sharding?

They probably will! 2-3x is kind of apple-apple comparison on the same number of buildkite sharding
